File: gdx-gltf-starter/core/src/com/mygdx/game/terrains/TerrainMaterial.java
package com.mygdx.game.terrains;

import com.mygdx.game.terrains.attributes.TerrainAttributes;

public class TerrainMaterial extends TerrainAttributes {
    private static int counter = 0;

    public String id;

    /**
     * Create an empty material
     */
    public TerrainMaterial() {
        this("mtl" + (++counter));
    }

    /**
     * Create an empty material
     */
    public TerrainMaterial(final String id) {
        this.id = id;
    }

    @Override
    public int hashCode() {
        return super.hashCode() + 3 * id.hashCode();
    }

    @Override
    public boolean equals(Object other) {
        return (other instanceof TerrainMaterial) && ((other == this) || ((((TerrainMaterial) other).id.equals(id)) && super.equals(other)));
    }
}

File: GDX_GLTF_Example/core/src/com/mygdx/game/terrains/TerrainMaterial.kt
package com.mygdx.game.terrains

import com.mygdx.game.terrains.attributes.TerrainAttributes

class TerrainMaterial
/**
 * Create an empty material
 */ @JvmOverloads constructor(var id: String = "mtl" + (++counter)) : TerrainAttributes() {
    /**
     * Create an empty material
     */

    override fun hashCode(): Int {
        return super.hashCode() + 3 * id.hashCode()
    }

    override fun equals(other: Any?): Boolean {
        return (other is TerrainMaterial) && ((other === this) || ((other.id == id) && super.equals(other)))
    }

    companion object {
        private var counter = 0
    }
}
